adr-11-centralized-sequencer.md

```diff
@@ -0,0 +1,213 @@
+# ADR 11: Centralized Sequencer
+
+## Changelog
+
+- 2024-10-01: Initial draft
+
+## Context
+
+Rollkit supports modular sequencer implementations and a centralized sequencer is a simple and efficient solution that can serve as a starting point for rollup developers who don't need the complexity of a decentralized sequencing solution.
+
+The centralized sequencer needs to implement the Generic Sequencer interface defined in the `go-sequencing` package, provide transaction batching capabilities, and reliably submit these batches to a DA layer. It should also maintain state to track submitted batches and provide verification capabilities.
+
+## Alternative Approaches
+
+### Decentralized Sequencer
+
+A decentralized sequencer would distribute the sequencing responsibility across multiple nodes, providing better censorship resistance and fault tolerance. However, this approach introduces significant complexity in terms of consensus, leader election, and coordination between nodes. It would also require more resources to operate and maintain.
+
+This approach was not chosen for the initial implementation because:
+1. It adds unnecessary complexity for many use cases
+2. It requires more development time and resources
+3. Many rollup projects start with a centralized sequencer and gradually move towards decentralization
+
+### Embedded Sequencer in Rollup Nodes
+
+Another approach would be to embed sequencing functionality directly into rollup nodes. This would simplify the architecture by eliminating a separate sequencer component.
+
+This approach was not chosen because:
+1. It couples sequencing logic with rollup node logic, reducing modularity
+2. It makes it harder to upgrade or replace the sequencing component independently
+3. It doesn't allow for a dedicated sequencing service that can be optimized separately
+
+## Decision
+
+We implement a standalone centralized sequencer that:
+
+1. Implements the Generic Sequencer interface from the `go-sequencing` package
+2. Exposes a gRPC service for rollup clients to submit transactions
```

Comment:
this isnt the case with execution api anymore right?

Reply:
Yes, not the case once we move this code back into Rollkit